hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/physical/CreateDagVisitor.java

```diff
@@ -113,10 +121,40 @@ public Vertex onFilter(FilterPhysicalRel rel) {
                 ServiceFactories.nonSharedService(ctx ->
                         ExpressionUtil.filterFn(filter, SimpleExpressionEvalContext.from(ctx))),
                 (BiPredicateEx<Predicate<Object[]>, Object[]>) Predicate::test));
-        connectInput(rel.getInput(), vertex, null);
+        connectInputPreserveCollation(rel, vertex);
         return vertex;
     }
 
+    public Vertex onSort(SortPhysicalRel rel) {
+        ComparatorEx<?> comparator = ExpressionUtil.comparisonFn(rel.getCollations());
+
+        Vertex sortVertex = dag.newUniqueVertex("Sort",
+                ProcessorMetaSupplier.of(
+                        sortP(comparator)
+                )
+        );
+        connectInput(rel.getInput(), sortVertex, null);
+
+        Vertex combineVertex = dag.newUniqueVertex("SortCombine",
+                ProcessorMetaSupplier.forceTotalParallelismOne(
+                        ProcessorSupplier.of(
+                                mapP(FunctionEx.identity())
+                        ),
+                        localMemberAddress
+                )
+        );
+        connectInput(
```

Comment:
I would connect this edge directly by inlining this method. The connections between multiple vertices created from a single rel should be connected like that. The `connectInput` method is meant to recurse the visitor to the input rel and connect the vertex produced by the input rel.

Reply:
I refactored out the common logic into a new `connectInput(Vertex, Vertex, ...)` method. Do you want me to remove it also?